request/do.go

```diff
@@ -32,11 +33,22 @@ func doRequest(url string, timeout time.Duration) record.Record {
 // The value of concurrency limits the number of concurrent threads.
 // Once all requests have been made or on done signal from ctx,
 // waits for goroutines to end and returns the collected records.
-func Do(ctx context.Context, requests, concurrency int, url string, timeout time.Duration) []record.Record {
-	records := record.NewSafeSlice(requests)
-
-	semimpl.Do(ctx, concurrency, requests, func() {
-		records.Append(doRequest(url, timeout))
+func Do(cfg config.Config) []record.Record {
+	var (
+		uri        = cfg.Request.URL.String()
+		numWorker  = cfg.RunnerOptions.Concurrency
+		numRequest = cfg.RunnerOptions.Requests
+		reqTimeout = cfg.Request.Timeout
+		gloTimeout = cfg.RunnerOptions.GlobalTimeout
```

Comment:
Ew that's not sexy.

I really think `duration` was better, one of the reason is exactly this one, struggling to differentiate things with the same name by adding more characters to it in unsatisfying ways.

`duration` really means "the entire duration the benchmark should run for, leave blank to default to whatever we choose was the acceptable max duration other than infinity".

Reply:
This corresponds to the specs we agreed one. We can still re-discuss them, but I suggest not here.

Reply:
I'm fine with that for now but i think we can still re-discuss the specs here as they are fuzzy at best. See:

> It does not follow all the specs from the issue: it will be properly updated and the current one will be rescoped to a MVP milestone.

Reply:
What I meant here is that I didn't implement _all_ the features requested. Not that I would do just as it pleases me.
Not discussing that here.